sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample08_EventProcessor.md

```diff
@@ -0,0 +1,230 @@
+                    BlobContentInfo info = await ownershipBlobClient.UploadAsync(emptyStream, metadata: ownershipMetadata, conditions: requestConditions, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                    claimedOwnerships.Add(new EventProcessorPartitionOwnership()
+                    {
+                        ConsumerGroup = ownership.ConsumerGroup,
+                        EventHubName = ownership.EventHubName,
+                        FullyQualifiedNamespace = ownership.FullyQualifiedNamespace,
+                        LastModifiedTime = info.LastModified,
+                        OwnerIdentifier = ownership.OwnerIdentifier,
+                        PartitionId = ownership.PartitionId,
+                        Version = info.ETag.ToString()
+                    });
+                }
+                else
+                {
+                    // In this case, the partition is owned by some other host. The ownership file already exists, so we just need to change metadata on it. But we should only do this if the metadata has not
+                    // changed between when we listed ownership and when we are trying to claim ownership, i.e. the ETag for the file has not changed.               
+                    BlobRequestConditions requestConditions = new BlobRequestConditions() { IfMatch = new ETag(ownership.Version) };
+                    BlobInfo info = await ownershipBlobClient.SetMetadataAsync(ownershipMetadata, requestConditions, cancellationToken).ConfigureAwait(false);
+
+                    claimedOwnerships.Add(new EventProcessorPartitionOwnership()
+                    {
+                        ConsumerGroup = ownership.ConsumerGroup,
+                        EventHubName = ownership.EventHubName,
+                        FullyQualifiedNamespace = ownership.FullyQualifiedNamespace,
+                        LastModifiedTime = info.LastModified,
+                        OwnerIdentifier = ownership.OwnerIdentifier,
+                        PartitionId = ownership.PartitionId,
+                        Version = info.ETag.ToString()
+                    });
+                }
+            }
+            catch (RequestFailedException e) when (e.ErrorCode == BlobErrorCode.BlobAlreadyExists || e.ErrorCode == BlobErrorCode.ConditionNotMet)
+            {
+                // In this case, another host has claimed the partition before we did. That's safe to ignore. We'll still try to claim other partitions.
+            }
+        }
+
+        return claimedOwnerships;
+    }
+
+    private const string CheckpointPrefixFormat = "{0}/{1}/{2}/checkpoint/";
+    private const string OffsetMetadataKey = "offset";
+
+    // We use the same strategy for recording checkpoint information as ownership information (metadata on a blob in blob storage)
+    protected override async Task<IEnumerable<EventProcessorCheckpoint>> ListCheckpointsAsync(CancellationToken cancellationToken = default)
```

Comment:
Storage Reference:  [GetCheckpointAsync](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobCheckpointStore/BlobsCheckpointStore.cs#L363)
Processor Reference: [GetCheckpointAsync](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs#L797)